buildengine/watch.go

```diff
@@ -30,17 +32,42 @@ type WatchEventProjectChanged struct {
 
 func (WatchEventProjectChanged) watchEvent() {}
 
+type projectHashes struct {
+	Hashes  FileHashes
+	Project Project
+}
+
+type WatchService struct {
+	isWatching bool
+
+	// use mutex whenever accessing / modifying existingProjects or moduleTransactions
+	mutex              sync.Mutex
+	existingProjects   map[string]projectHashes
+	moduleTransactions map[string][]*ModifyFilesTransaction
+}
+
+func NewWatchService(ctx context.Context) *WatchService {
```

Comment:
Also why not start + return the Watcher in the constructor? Or are we doing something between constructing and watching somewhere?

Reply:
Here's what I was trying to balance between:
- When running `ftl dev` we want the watcher to be actively watching. Otherwise there is nothing to watch.
- I don't want the controller / build systems to care whether the transaction does anything.
- So the current implementation is we always have a watcher which can provide transactions, but only starts watching if required.